### PR TITLE
Adding new features on IndiceCreation Page

### PR DESCRIPTION
**Pull request by AymericNoel**

### CODE REVIEW

**Comment by AymericNoel** [comment]
To do in next PR : 
+  save initial valuation in database
+  update index button should change name/symbol/description of index in db, including in its saved compositions

**Review by monsieurleberre — COMMENTED** [review]
I didn't check the code in details but tried it out a bit and this looks weird, the nav is very far from the targeted nav...
image

File: src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceCreation.razor [hunk 1]
```diff
@@ -557,4 +607,7 @@
         }
     }
     #endregion
+
+
+
```
Comment (monsieurleberre) [inline src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceCreation.razor:612]: damn, this file is now 600 lines of code, I think it woudl be a good exercise to split it into re-usable components, or extract some functions into dedicated services. Not too sure how is the best way to do that now, but let's keep it in mind!

File: src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceList.razor [hunk 2]
```diff
@@ -31,33 +31,33 @@
...
+                                    <SfGrid DataSource="indice.IndiceCompositions" @ref="CompositionGrid" AllowResizing="true">
                                         <GridSelectionSettings EnableToggle="true" Type="SelectionType.Single"></GridSelectionSettings>
                                         <GridTemplates>
                                             <DetailTemplate Context="child_context">
                                                 @{
                                                     var composition = (child_context as IndiceCompositionModel);
                                                     <h4>@composition.Symbol Components</h4>
                                                     <div class="row">
-                                                        <SfGrid DataSource="composition.Components" >
+                                                        <SfGrid DataSource="composition.Components" AllowResizing="true">
```
Comment (monsieurleberre) [inline src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceList.razor:42]: does it naturally put it in a configuration that shows everything, or do you always end up having to resize?

File: src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceCreation.razor (hunk 1 above)
Comment (AymericNoel) [reply src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceCreation.razor:612]: Yeah this file is a monster ....

File: src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceList.razor (hunk 2 above)
Comment (AymericNoel) [reply src/Trakx.IndiceManager.Client/Pages/IndiceComposition/IndiceList.razor:42]: Generally, it shows everything, this is an extra for users who want to customize the displaying.

**Review by jmc171144 — COMMENTED** [review]
Looks good 👏 missing some tests but everything will be perfect then!

File: src/Trakx.Common/Interfaces/IIndiceDataProvider.cs [hunk 3]
```diff
@@ -127,8 +127,9 @@ public interface IIndiceDataProvider
         /// <summary>
         /// Returns a composition with a given Id.
         /// </summary>
-        /// <param name="compositionId">The id of the composition that we're looking for.</param>
+        /// <param name="indexSymbol">The symbol of the index of the composition that we're looking for.</param>
+        /// <param name="version">The version of the composition that we're looking for.</param>
         /// <returns>Returns the composition associated to the id, null else.</returns>
-        Task<IIndiceComposition?> GetCompositionById(string compositionId);
+        Task<IIndiceComposition?> GetCompositionBySymbolAndVersion(string indexSymbol,uint version);
```
Comment (monsieurleberre) [inline src/Trakx.Common/Interfaces/IIndiceDataProvider.cs:133]: 👍

**Review by monsieurleberre — APPROVED** [review]
Thanks, let's merge!!

**Pull request merged**